### PR TITLE
Added runners to register and deploy ml-model

### PR DESCRIPTION
**Pull request by vpehkone**
### Description
Added runners DeletePipeline, DeleteMlModel, RegisterMlModel and DeployMlModel. So that we can run the new workload semantic_search, that does indexing and search with vector embedding.

### Issues Resolved
https://github.com/opensearch-project/opensearch-benchmark-workloads/issues/198

### Testing
Tested by executing the neural_search workload

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/worker_coordinator/runner.py [hunk 1]
```diff
@@ -1272,6 +1277,20 @@ async def __call__(self, opensearch, params):
...
+class DeletePipeline(Runner):
+    @time_func
+    async def __call__(self, opensearch, params):
+        try:
+            await opensearch.ingest.delete_pipeline(id=mandatory(params, "id", self),
+                                                    master_timeout=params.get("master-timeout"),
+                                                    timeout=params.get("timeout"),
+                                                    )
+        except:
+            pass # no current pipeline
```
Comment (IanHoang) [inline osbenchmark/worker_coordinator/runner.py:1289]: Might be useful to still add a logging statement that states that there is no current pipeline to delete before swallowing the exception.

File: osbenchmark/worker_coordinator/runner.py [hunk 2]
```diff
@@ -2358,3 +2377,116 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
...
+        try:
+            resp = await opensearch.transport.perform_request('POST', '/_plugins/_ml/models/_search', body=body)
+            for item in resp['hits']['hits']:
+                doc = item.get('_source')
+                if doc:
+                    id = doc.get('model_id')
+                    if id:
+                        model_ids.add(id)
+        except:
+            pass # no current model
```
Comment (IanHoang) [inline osbenchmark/worker_coordinator/runner.py:2405]: Recommend adding a logging statement that there is no current model before swallowing

File: osbenchmark/worker_coordinator/runner.py [hunk 3]
```diff
@@ -2358,3 +2377,116 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
...
+        try:
+            resp = await opensearch.transport.perform_request('POST', '/_plugins/_ml/models/_search', body=search_body)
+            for item in resp['hits']['hits']:
+                doc = item.get('_source')
+                if doc:
+                    model_id = doc.get('model_id')
+                    if model_id:
+                        break
+        except:
+            pass
```
Comment (IanHoang) [inline osbenchmark/worker_coordinator/runner.py:2445]: Recommend adding logging statement for exception here.

File: osbenchmark/worker_coordinator/runner.py [hunk 4]
```diff
@@ -2358,3 +2377,116 @@ async def __aexit__(self, exc_type, exc_val, exc_tb):
...
+                "version": params.get('model-version'),
+                "model_format": params.get('model-format')
+            }
+        search_body = {
+            "query": {
+                "match": {
+                    "name": body['name']
+                }
+            },
+            "size": 1000
```
Comment (IanHoang) [inline osbenchmark/worker_coordinator/runner.py:2433]: Will we ever need to change the size parameter? If so, might be useful to have a variable that references its value

File: osbenchmark/worker_coordinator/runner.py (hunk 3 above)
Comment (IanHoang) [reply osbenchmark/worker_coordinator/runner.py:2445]: Pylint also suggests that this is a bare except.
```
osbenchmark/worker_coordinator/runner.py:1288:8: W0702: No exception type(s) specified (bare-except)
osbenchmark/worker_coordinator/runner.py:2404:8: W0702: No exception type(s) specified (bare-except)
osbenchmark/worker_coordinator/runner.py:2444:8: W0702: No exception type(s) specified (bare-except)
```

File: osbenchmark/worker_coordinator/runner.py (hunk 2 above)
Comment (IanHoang) [reply osbenchmark/worker_coordinator/runner.py:2405]: Pylint also suggests that this is a bare except.
```
osbenchmark/worker_coordinator/runner.py:1288:8: W0702: No exception type(s) specified (bare-except)
osbenchmark/worker_coordinator/runner.py:2404:8: W0702: No exception type(s) specified (bare-except)
osbenchmark/worker_coordinator/runner.py:2444:8: W0702: No exception type(s) specified (bare-except)
```

File: osbenchmark/worker_coordinator/runner.py (hunk 1 above)
Comment (IanHoang) [reply osbenchmark/worker_coordinator/runner.py:1289]: Pylint also suggests that this is a bare except.
```
osbenchmark/worker_coordinator/runner.py:1288:8: W0702: No exception type(s) specified (bare-except)
osbenchmark/worker_coordinator/runner.py:2404:8: W0702: No exception type(s) specified (bare-except)
osbenchmark/worker_coordinator/runner.py:2444:8: W0702: No exception type(s) specified (bare-except)
```

**Review by IanHoang — CHANGES_REQUESTED** [review]
Please add an exception class and logging statement to except clauses

**Comment by vpehkone** [comment]
@IanHoang Can you please review, I have addressed the issues.

**Review by IanHoang — APPROVED** [review]

**Pull request merged**